metrics/local_sink.go

```diff
@@ -151,16 +157,22 @@ func (sink *localSink) Flush() error {
 		switch metric := i.(type) {
 		case _metrics.Counter:
 			if shouldFlush(metricTypeCounter, name) {
+				sink.s.flushed++
+				counters++
 				sink.dst.Handle(metricName, tags, float64(metric.Count()), metricTypeGauge)
 			}
 		case _metrics.GaugeFloat64:
 			if shouldFlush(metricTypeGauge, name) {
+				sink.s.flushed++
+				gauges++
 				sink.dst.Handle(metricName, tags, float64(metric.Value()), metricTypeGauge)
 			}
 		case _metrics.Histogram:
-			h := metric.Snapshot()
+			h := metric.Sample()
```

Comment:
Notice that this is different. Does it actually matter if we don't use the snapshot and instead use the Sample() directly given that it has mutex.

Reply:
@i0exception